pkg/segment/tracing/handler/traceHandler.go

```diff
@@ -0,0 +1,219 @@
+		} else {
+			log.Errorf("ProcessSearchTracesRequest: request does not contain required parameter: searchText")
+			return
+		}
+
+		pipeSearchResponseOuter, err := processSearchRequest(searchRequestBody, myid)
+		if err != nil {
+			log.Errorf("ProcessSearchTracesRequest: %v", err)
+			return
+		}
+		traceIds = GetUniqueTraceIds(pipeSearchResponseOuter, startEpoch, endEpoch)
+	}
+
+	traces := make([]*structs.Trace, 0)
+	// Get status code count for each trace
+	for _, traceId := range traceIds {
+		// Get the start time and end time for this trace
+		searchRequestBody.SearchText = "trace_id=" + traceId + " AND parent_span_id=\"\" | fields start_time, end_time"
+		pipeSearchResponseOuter, err := processSearchRequest(searchRequestBody, myid)
+		if err != nil {
+			log.Errorf("ProcessSearchTracesRequest: traceId:%v, %v", traceId, err)
+			continue
+		}
+
+		if pipeSearchResponseOuter.Hits.Hits == nil || len(pipeSearchResponseOuter.Hits.Hits) == 0 {
+			continue
+		}
+
+		startTime, exists := pipeSearchResponseOuter.Hits.Hits[0]["start_time"]
+		if !exists {
+			continue
+		}
+		endTime, exists := pipeSearchResponseOuter.Hits.Hits[0]["end_time"]
+		if !exists {
+			continue
+		}
+
+		traceStartTime := uint64(startTime.(float64))
+		traceEndTime := uint64(endTime.(float64))
+
+		// Only process traces which start and end in this period [startEpoch, endEpoch]
+		if (startEpoch*1e6 > traceStartTime) || (endEpoch*1e6 < traceEndTime) {
+			continue
+		}
+
+		searchRequestBody.SearchText = "trace_id=" + traceId + " | stats count BY status_code"
+		pipeSearchResponseOuter, err = processSearchRequest(searchRequestBody, myid)
+		if err != nil {
+			log.Errorf("ProcessSearchTracesRequest: traceId:%v, %v", traceId, err)
+			continue
+		}
+
+		AddTrace(pipeSearchResponseOuter, &traces, traceId, traceStartTime, traceEndTime)
+	}
+
+	traceResult := &structs.TraceResult{
+		Traces: traces,
+	}
+
+	utils.WriteJsonResponse(ctx, traceResult)
+	ctx.SetStatusCode(fasthttp.StatusOK)
+}
+
+func GetUniqueTraceIds(pipeSearchResponseOuter *pipesearch.PipeSearchResponseOuter, startEpoch uint64, endEpoch uint64) []string {
+	traceIds := make([]string, 0)
+	for _, bucket := range pipeSearchResponseOuter.Aggs[""].Buckets {
+		traceId, exists := bucket["key"]
+		if !exists {
+			continue
+		}
+		traceIds = append(traceIds, traceId.(string))
+	}
+	return traceIds
+}
+
+// Check if searchText only contains traceId as query condition
+func ExtractTraceID(searchText string) (bool, string) {
+	pattern := `^trace_id=([a-zA-Z0-9]+)$`
+
+	regex, err := regexp.Compile(pattern)
+	if err != nil {
+		return false, ""
+	}
+
+	matches := regex.FindStringSubmatch(searchText)
+	if len(matches) != 2 {
+		return false, ""
+	}
+
+	return true, matches[1]
+}
+
+func AddTrace(pipeSearchResponseOuter *pipesearch.PipeSearchResponseOuter, traces *[]*structs.Trace, traceId string, traceStartTime uint64, traceEndTime uint64) {
```

Comment:
I think you should be able to do `traces []*structs.Trace` instead of `traces *[]*structs.Trace` (but I might be wrong about this) since slices are passed by reference. But you can keep what you have

Reply:
Yes, I use `&` to pass reference. After modify `traces []*structs.Trace` instead of traces `*[]*structs.Trace`, it gives me an error and  i can not use `&` to pass traces. So i will keep it.